vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -181,7 +203,7 @@ func (r RefCountsMap) Init(d drivers.VolumeDriver, mountDir string, name string)
 	err = r.discoverAndSync(c, d)
```

Comment:
I am a bit skeptical about failing unmount requests while recalc is in flight. The reason being docker ignores failures in unmount. I tried this with a small test.

What we can do is as follows:
While retrying to populate refCount: do it in the reconcile map. The reason being that talking to docker using engine api might be consuming. So we allow parallel mount unmounts during this time.

Once we have reconcile-refmap populated, swap this with original refCount under the refCount mutex. Also sync the mount points  and remove unwanted mounts while maintaing the mutex. After this is done, release the mutex. Since the mount and unmount request too access the refcount, this mutex will hang those requests for a bit.
I understand that using a mutex like this doesn’t principally sound well because unmounting volumes might take a while.

But if we are gonna fail mount/unmount, we might as well hang them for a while. 
CC @govint